lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1193,15 +1193,23 @@ def _create_node_runbook(
             azure_node_runbook.marketplace = self._resolve_marketplace_image(
                 azure_node_runbook.location, azure_node_runbook.marketplace
             )
-            image_info = self._get_image_info(
-                azure_node_runbook.location, azure_node_runbook.marketplace
-            )
-            # HyperVGenerationTypes return "V1"/"V2", so we need to strip "V"
-            azure_node_runbook.hyperv_generation = _get_vhd_generation(image_info)
+            image_info = None
+            try:
+                image_info = self._get_image_info(
+                    azure_node_runbook.location, azure_node_runbook.marketplace
+                )
+                # HyperVGenerationTypes return "V1"/"V2", so we need to strip "V"
+                azure_node_runbook.hyperv_generation = _get_vhd_generation(image_info)
+            except Exception as e:
```

Comment:
Instead of add it every place the _get_image_info called. How about capturing the exception in this method? Other place just check if it's None or not. If it's None then skip some logic.